rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -605,11 +552,58 @@ def find_elements_by(how, what, parent = nil)
           ids.map { |id| Element.new self, element_id_from(id) }
         end
 
+        protected
+
+        def capabilities_class
+          W3CCapabilities
+        end
+
+        def default_capabilities
+          capabilities_class.firefox
+        end
+
         private
 
-        def assert_javascript_enabled
-          return if capabilities.javascript_enabled?
-          raise Error::UnsupportedOperationError, 'underlying webdriver instance does not support javascript'
+        def bridge_module
+          Module.nesting[1]
+        end
+
+        def process_capabilities(opts)
+          capabilities = opts.delete(:desired_capabilities) || default_capabilities
+          capabilities = Capabilities.send(capabilities) if capabilities.is_a?(Symbol)
```

Comment:
Should we merge passed in `desired_capabilities` to the defaults for that browser? We had to do this for firefox currently so that `browser_name` was picked up (geckodriver needed it).